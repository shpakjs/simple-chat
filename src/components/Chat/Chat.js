import React from 'react';
import styles from './Chat.module.css';
import publicChat from '../../assets/public.png';
import privateChat from '../../assets/private.png';

const Chat = ({ id, caption, type, isSelected, onChatSelected }) => {
    return (
        <div className={`${styles.chat} ${type === "1" ? styles.private: ''} ${isSelected ? styles.active : '' }`}
            id = {id} 
            onClick = { () => onChatSelected(id) }>
                <img className={ styles.photo } src={ type === "2" ? publicChat : privateChat  } alt="user" />
                <span className={ styles.name }>{ caption }</span>
        </div>
    );
}
export default Chat;