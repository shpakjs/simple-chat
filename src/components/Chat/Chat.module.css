.chat {
    position: relative;
    font-size: 16px;    
    display: list-item;
    background: #fff;
    padding: 10px;
    border-bottom: 1px solid #ddd;
}
.chat.active,
.chat.active .date {
    background: rgb(255, 255, 255);
    color: #fff;
    box-shadow: inset 0 0 11px 3px rgba(0, 43, 51, 0.1);
    transition: .2s ease-in-out;
    border: 1px solid #ccc;
    border-width: 1px 0;
} 

.photo {
    width: 50px;
    height: 50px;
} 
.name {
    margin-left: 10px;
    vertical-align: top;
    font-weight: 600;
    color: #444;
}

.private{
    background: #f5f5f5;
}