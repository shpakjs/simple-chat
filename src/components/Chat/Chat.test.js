// Link.react.test.js
import React from 'react';
import Chat from './Chat';
import {create} from 'react-test-renderer';

describe("Chat component", () => {

    test("callback should be called", () => {
        const mockCallback = jest.fn();
        const component = create(<Chat 
            id="2" 
            caption="I am test chat" 
            type="1" 
            isSelected="false" 
            onChatSelected={mockCallback}/>);
        let tree = component.toJSON();
        tree.props.onClick();
        expect(mockCallback.mock.calls.length).toBe(1);
    });

    test('Chat component changed on click', () => {
        const component = create(<Chat 
            id="2" 
            caption="I am test chat" 
            type="1" 
            isSelected="false" 
            onChatSelected={() => {}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        tree.props.onClick();
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

});
