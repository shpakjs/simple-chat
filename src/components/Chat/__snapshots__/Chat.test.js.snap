// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component Chat component changed on click 1`] = `
<div
  className="chat private active"
  id="2"
  onClick={[Function]}
>
  <img
    alt="user"
    className="photo"
    src="private.png"
  />
  <span
    className="name"
  >
    I am test chat
  </span>
</div>
`;

exports[`Chat component Chat component changed on click 2`] = `
<div
  className="chat private active"
  id="2"
  onClick={[Function]}
>
  <img
    alt="user"
    className="photo"
    src="private.png"
  />
  <span
    className="name"
  >
    I am test chat
  </span>
</div>
`;
